pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -886,6 +894,118 @@ LANGUAGE SQL VOLATILE;
 COMMENT ON FUNCTION SCHEMA_PROM.reset_metric_retention_period(TEXT)
 IS 'resets the retention period for a specific metric to using the default';
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_metric_compression_setting(metric_name TEXT)
+RETURNS BOOLEAN
+AS $$
+DECLARE
+    can_compress boolean;
+    result boolean;
+    metric_table_name text;
+BEGIN
+    SELECT exists(select * from pg_proc where proname = 'compress_chunk')
+    INTO STRICT can_compress;
+
+    IF NOT can_compress THEN
+        RETURN FALSE;
+    END IF;
+
+    SELECT table_name 
+    INTO STRICT metric_table_name
+    FROM SCHEMA_CATALOG.get_or_create_metric_table_name(metric_name);
+
+    SELECT EXISTS ( 
+        SELECT FROM _timescaledb_config.bgw_policy_compress_chunks p
+        INNER JOIN _timescaledb_catalog.hypertable h 
+        ON (h.id = p.hypertable_id) 
+        WHERE h.schema_name = 'SCHEMA_DATA' 
+        AND h.table_name = metric_table_name)
+    INTO result;
+    RETURN result;
+END
+$$
+LANGUAGE PLPGSQL STABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.get_metric_compression_setting(TEXT) TO prom_reader;
+
+CREATE OR REPLACE FUNCTION SCHEMA_PROM.set_default_compression_setting(compression_setting BOOLEAN)
```

Comment:
This doesn't change the setting on existing tables. I do think we need to add a `default_compression BOOLEAN NOT NULL DEFAULT true` to `SCHEMA_CATALOG.metric` similar to `default_chunk_interval` to tell which metrics follow the default and which do not.